test/integration/PermissionTable.test.ts

```diff
@@ -0,0 +1,218 @@
+  // [ 'OPTIONS', 'C/R', [],                     undefined,              '',     '',  401, 401 ],
+  // [ 'OPTIONS', 'C/R', [],                     [ AM.read ],            '',     '',  200, 404 ],
+  // [ 'OPTIONS', 'C/R', [ AM.read ],            undefined,              '',     '',  200, 404 ],
+
+  [ 'HEAD',    'C/R', [],                     undefined,              '',     '',  401, 401 ],
+  [ 'HEAD',    'C/R', [],                     [ AM.read ],            '',     '',  200, 404 ],
+  [ 'HEAD',    'C/R', [ AM.read ],            undefined,              '',     '',  200, 404 ],
+
+  [ 'GET',     'C/R', [],                     undefined,              '',     '',  401, 401 ],
+  [ 'GET',     'C/R', [],                     [ AM.read ],            '',     '',  200, 404 ],
+  [ 'GET',     'C/R', [ AM.read ],            undefined,              '',     '',  200, 404 ],
+  // For the 404, we only check if C/R has read permissions, not C/
+  // [ 'GET',     'C/R', [ AM.read ],            [ AM.write ],           '',     '',  401, 404 ],
+
+  [ 'POST',    'C/',  [],                     undefined,              '',     TXT, 401, 401 ],
+  [ 'POST',    'C/',  [],                     [ AM.read ],            '',     TXT, 401, 401 ],
+  [ 'POST',    'C/',  [ AM.append ],          undefined,              '',     TXT, 201, 201 ],
+  [ 'POST',    'C/',  [ AM.append ],          [ AM.read ],            '',     TXT, 201, 201 ],
+  [ 'POST',    'C/',  [ AM.read, AM.append ], undefined,              '',     TXT, 201, 201 ],
+  [ 'POST',    'C/',  [ AM.read, AM.append ], [ AM.read ],            '',     TXT, 201, 201 ],
+
+  [ 'PUT',     'C/',  [],                     undefined,              '',     N3,  401, 401 ],
+  [ 'PUT',     'C/',  [ AM.read ],            undefined,              '',     N3,  401, 401 ],
+  [ 'PUT',     'C/',  [ AM.write ],           undefined,              '',     N3,  205, 201 ],
+
+  [ 'PUT',     'C/R', [],                     undefined,              '',     TXT, 401, 401 ],
+  [ 'PUT',     'C/R', [],                     [ AM.read ],            '',     TXT, 401, 401 ],
+  [ 'PUT',     'C/R', [],                     [ AM.append ],          '',     TXT, 401, 401 ],
+  [ 'PUT',     'C/R', [],                     [ AM.write ],           '',     TXT, 205, 401 ],
+  [ 'PUT',     'C/R', [ AM.read ],            undefined,              '',     TXT, 401, 401 ],
+  [ 'PUT',     'C/R', [ AM.append ],          undefined,              '',     TXT, 401, 401 ],
+  [ 'PUT',     'C/R', [ AM.write ],           undefined,              '',     TXT, 205, 201 ],
+  [ 'PUT',     'C/R', [ AM.append ],          [ AM.write ],           '',     TXT, 205, 201 ],
+
+  [ 'PATCH',  'C/R', [],                     undefined,              DELETE,  N3,  401, 401 ],
+  // We don't return 404 yet in case a PATCH has no inserts and C/R does not exist
+  // [ 'PATCH',   'C/R', [],                     [ AM.read ],            DELETE,  N3,  401, 404 ],
```

Comment:
This is the extension to the integration test setup to cover it, problem is that we fail this test. 😄

Reply:
Ah I had understood "don't return yet" as in "the server determines this at a later stage.

I have uncommented this line and it works…

92 (similar comment) however fails with a 409 rather than a 404. Not the best, but not a problem I think.

Reply:
For the 409, the solution would be to always check in the authorizer if a) we have read permission, b) the request does not require create permission and, c) the resource does not exist.

Currently we only check this if there would be an authorization error. This would make it so all GET requests now trigger a `hasResource` call though.